src/modules/firewall/tests/FirewallTests.cpp

```diff
@@ -8,52 +8,149 @@
 
 namespace tests
 {
-    static const std::string expectedFingerprint = "abc123";
-    static const std::string expectedFingerprintJson = "\"" + expectedFingerprint + "\"";
+    testing::AssertionResult JsonEq(const std::string& expectedJson, const std::string& actualJson)
+    {
+        rapidjson::Document actual;
+        rapidjson::Document expected;
 
-    class MockUtility : public GenericFirewall
+        if (expected.Parse(expectedJson.c_str()).HasParseError())
+        {
+            return testing::AssertionFailure() << "expected JSON is not valid JSON" << expectedJson;
+        }
+        else if (actual.Parse(actualJson.c_str()).HasParseError())
+        {
+            return testing::AssertionFailure() << "actual JSON is not valid JSON" << actualJson;
+        }
+        else if (actual == expected)
+        {
+            return testing::AssertionSuccess();
+        }
+        else
+        {
+            return testing::AssertionFailure() << "expected:\n\t" << expectedJson << "\n but got:\n\t" << actualJson;
+        }
```

Comment:
Rather than many exists from function, assign to some internal variable from if-else and return once.